atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/util/MeasuringUtils.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.util;
```

Comment:
Will leave for the next reviewer: this is kind of an interesting problem, as java doesn't let use generics and concrete types for method overloading (so Map<? extends Measurable, ? extends Measurable> cannot have the same method for Map<? extends Measurable, Long>). This results in the numerous naming of methods to below.

However, I don't see an ideal solution out of this outside of potentially splitting this class up (or potentially wrapping primitive, but that would introduce api breaks and be jvm intensive as we'd need to wrap every long/byte array, etc) .